File: Movie-Catalog-Service/src/main/java/com/example/Movie_Catalog_Service/model/Movie.java
package com.example.Movie_Catalog_Service.model;



import com.google.cloud.Timestamp;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Movie {
    private String movieId;
    private String title;
    private String description;
    private String posterUrl;
    private String backdropUrl;
    private String trailerUrl;
    private int durationMinutes;
    private List<String> genre;
    private Timestamp releaseDate;
    private String director;
    private List<String> cast;
}

File: AndroidApp/app/src/main/java/com/btlcnpm/androidapp/data/model/Movie.kt
package com.btlcnpm.androidapp.data.model

// Data class cho response từ Movie Catalog Service
data class Movie(
    val movieId: String?,
    val title: String?,
    val description: String?,
    val posterUrl: String?, // Dùng cho Coil
    val backdropUrl: String?,
    val trailerUrl: String?,
    val durationMinutes: Int?,
    val genre: List<String>?,
    val releaseDate: String?, // Đã sửa lỗi: Backend trả về String ISO 8601
    val director: String?,
    val cast: List<String>?
)

// DTO cho Search (Dùng để nhất quán)
data class MovieSearchDTO(
    val movieId: String?,
    val title: String?,
    val posterUrl: String?
)